pkg/lib/feature/accountanonymization/runnable.go

```diff
@@ -38,7 +38,8 @@ func (r *Runnable) Run(ctx context.Context) error {
 		return err
 	}
 	for _, appUser := range appUsers {
-		appContext, err := r.AppContextResolver.ResolveContext(ctx, appUser.AppID)
+		var appContext *config.AppContext
+		ctx, appContext, err = r.AppContextResolver.ResolveContext(ctx, appUser.AppID)
```

Comment:
This is problematic. We are passing the project-specific ctx to ResolveContext in non-1st iteration of the loop.

It is better to extract the loop body into a separate function so that it takes a context.Context (the original one from Run()).